Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -181,23 +328,29 @@ private string GetZipURLFromJSON(JObject requestObject)
             }
         }
 
-        private async Task<HttpResponseMessage> PushDeployAsync(ZipDeploymentInfo deploymentInfo, bool isAsync)
+        private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo deploymentInfo, bool isAsync, JObject requestObject = null, string artifactType = Constants.Zip)
         {
             var content = Request.Content;
             var isRequestJSON = content.Headers?.ContentType?.MediaType?.Equals("application/json", StringComparison.OrdinalIgnoreCase);
             if (isRequestJSON == true)
             {
                 try
                 {
-                    var requestObject = await Request.Content.ReadAsAsync<JObject>();
-                    deploymentInfo.ZipURL = ArmUtils.IsArmRequest(Request) ? GetZipURLFromARMJSON(requestObject) : GetZipURLFromJSON(requestObject);
+                    // Read the request body if it hasn't been read already
+                    if (requestObject == null)
+                    {
+                        requestObject = await Request.Content.ReadAsAsync<JObject>();
+                    }
+                    deploymentInfo.RemoteURL = ArmUtils.IsArmRequest(Request) ? GetArticfactURLFromARMJSON(requestObject) : GetArtifactURLFromJSON(requestObject);
                 }
                 catch (Exception ex)
                 {
                     return ArmUtils.CreateErrorResponse(Request, HttpStatusCode.BadRequest, ex);
                 }
             }
-            else
+            // For zip artifacts (zipdeploy, wardeploy, onedeploy with type=zip), copy the request body in a temp zip file.
+            // It will be extracted later as part of the next steps
```

Comment:
// For zip artifacts (zipdeploy, wardeploy, onedeploy with type=zip), copy the request body in a temp zip file.
            // It will be extracted to the appropriate directory by the Fetch handler